java/src/main/java/com/google/appengine/tools/pipeline/impl/PipelineManager.java

```diff
@@ -473,6 +473,67 @@ public static void acceptPromisedValue(String promiseHandle, Object value)
     backEnd.save(updateSpec, generatorJob.getQueueSettings());
   }
 
+  public static Slot getPromisedValueSlot(String promiseHandle) {
```

Comment:
Most of it is a copy/paste from acceptPromisedValue. Please refactor to make acceptPromisedValue call this method.